composer.json

```
@@ -33,15 +33,12 @@
         "felixfbecker/language-server-protocol": "^1.5.2",
         "fidry/cpu-core-counter": "^0.4.1 || ^0.5.1 || ^1.0.0",
         "netresearch/jsonmapper": "^1.0 || ^2.0 || ^3.0 || ^4.0",
-        "nikic/php-parser": "^4.16",
+        "nikic/php-parser": "^5.0.0",
```

This is a dependency that Psalm shares with other tools like PHPUnit. In order to allow downstream projects to upgrade their dependencies gracefully, Psalm should produce a release first that is compatible with both major versions.

```suggestion
        "nikic/php-parser": "^4.18 || ^5.0",
```

---

I'm open to trying to create a version of this PR that would land on the `5.x` branch that would work with both parser v4 and v5. I think the main thing would be going back to the deprecated node names (eg `Encapsed` vs `Interpolated`), which are still supported in v5 via class aliases.

The thing I'd worry about is additional complexity, maintenance, and testing involved in supporting two versions. The differences are non-trivial, as evidenced by this PR. PHPStan also looks like it's planning to jump to parser v5 on its 2.0 release, with its 1.x releases only supporting parser v4.

I'd like feedback from the maintainers before making any major changes to this PR or creating a new one.